src/borg/remote.py

```diff
@@ -636,6 +636,12 @@ def borg_cmd(self, args, testing):
                 opts.append('--critical')
             else:
                 raise ValueError('log level missing, fix this code')
+            try:
+                borg_logger = logging.getLogger('borg')
+                if borg_logger.json:
+                    opts.append('--log-json')
```

Comment:
It sets the `json` attribute on the Borg logger in the serve process, which would be used to correctly use the progress indicator in check/replay/commit. However, remote --progress was broken some time ago with IIRC implied logging setup. So that has to be revisited anyway.